userspace/libsinsp/test/sinsp.ut.cpp

```diff
@@ -36,3 +37,91 @@ TEST(sinsp, external_event_processor_initialization)
 	EXPECT_EQ(my_sinsp.get_external_event_processor(), &processor);
 }
 
+class sinsp_with_test_input : public ::testing::Test {
+protected:
+	void SetUp() override
+	{
+		m_test_data = std::unique_ptr<scap_test_input_data>(new scap_test_input_data);
+		m_test_data->event_count = 0;
+		m_test_data->events = nullptr;
+	}
+
+	void TearDown() override
+	{
+		for (size_t i = 0; i < m_events.size(); i++)
+		{
+			free(m_events[i]);
+		}
+	}
+
+	sinsp inspector;
+
+	void open_inspector()
+	{
+		inspector.open_test_input(m_test_data.get());
+	}
+
+	scap_evt* add_event(uint64_t ts, uint64_t tid, enum ppm_event_type event_type, uint32_t n, ...)
+	{
+		struct scap_sized_buffer event_buf = {NULL, 0};
+		size_t event_size;
+		char error[SCAP_LASTERR_SIZE];
+		error[0] = '\0';
+
+		va_list args;
+		va_start(args, n);
+		int32_t ret = scap_event_encode_params_v(event_buf, &event_size, error, event_type, n, args);
+		va_end(args);
+
+		if(ret != SCAP_INPUT_TOO_SMALL) {
+			return nullptr;
+		}
+
+		event_buf.buf = malloc(event_size);
+		event_buf.size = event_size;
+
+		if(event_buf.buf == NULL) {
+			return nullptr;
+		}
+
+		va_start(args, n);
+		ret = scap_event_encode_params_v(event_buf, &event_size, error, event_type, n, args);
+		va_end(args);
+
+		if(ret != SCAP_SUCCESS) {
+			free(event_buf.buf);
+			event_buf.size = 0;
+			return nullptr;
+		}
+
+		scap_evt *event = static_cast<scap_evt*>(event_buf.buf);
+		m_events.push_back(event);
+		m_test_data->events = m_events.data();
+		m_test_data->event_count = m_events.size();
+
+		return event;
+	}
+
+	std::string get_field_as_string(sinsp_evt *evt, std::string field_name)
+	{
+		sinsp_filter_check *chk = g_filterlist.new_filter_check_from_fldname(field_name, &inspector, false);
+		chk->parse_field_name(field_name.c_str(), true, false);
+		std::string result = chk->tostring(evt);
+		return result;
+	}
+
+	unique_ptr<scap_test_input_data> m_test_data;
+	std::vector<scap_evt*> m_events;
+};
+
+TEST_F(sinsp_with_test_input, test_event)
+{
+	add_event(1657881958, 1, PPME_SYSCALL_OPEN_E, 3, "/tmp/the_file", 0, 0);
+
+	open_inspector();
+
+	sinsp_evt *evt;
+	inspector.next(&evt);
+	ASSERT_EQ(evt->get_type(), PPME_SYSCALL_OPEN_E);
+	ASSERT_EQ(get_field_as_string(evt, "evt.arg.name"), "/tmp/the_file");
+}
```

Comment:
Note for the reviewer: this is the interesting part. The way it works is that:
* every `add_event` call simply has all the fields used for a specific event
* `get_field_as_string` will simply return a field representation of the specific event, so we can check any field we wish (`fd.name`, `proc.name`, ...)
* We can add features to set an initial thread/fd state so we actually have a base to work on

Reply:
The possibility to get the specific event field is really powerful and I think this is exactly what we need here :)

Reply:
Obviously, this is just a test example but I propose to write some comments for every test we add, in which we try to explain the flow we are thinking about and what we expect from this test. I propose this because `libsinsp` internals are quite complex and coming back to test in the future could be not so easy to do without adequate documentation, WDYT about that?

Reply:
Do I read "networking fields"?
Yes absolutely, if the behavior is not obvious (such as opening a file and checking if `fd.name` is right) we can add comments. To make this even more readable we could interleave `add_event()` with `next()` calls as long as the processing is done on a single thread like it is today. That might not be extremely clean but it would show exactly the event we're processing and the expected state change or result.